pkg/registry/apis/provisioning/resources/parser.go

```diff
@@ -169,11 +170,15 @@ func (r *Parser) Parse(ctx context.Context, info *repository.FileInfo, validate
 
 	// Calculate name+folder from the file path
 	if info.Path != "" {
-		objName, folderName := NamesFromHashedRepoPath(r.repo.Name, info.Path)
-		parsed.Meta.SetFolder(folderName)
+		objName := FileNameFromHashedRepoPath(r.repo.Name, info.Path)
```

Comment:
in https://github.com/grafana/grafana/pull/103309 I think we should remove this option entirely for now

we have too many untested options where things may or may not have the name in the resource

Reply:
I challenged that part in that PR for the POST /files